roonapi/roonapi.py

```diff
@@ -233,15 +233,93 @@ def mute(self, output_id, mute=True):
         data = {"output_id": output_id, "how": how}
         return self._request(SERVICE_TRANSPORT + "/mute", data)
 
-    def change_volume(self, output_id, value, method="absolute"):
+    def set_volume_percent(self, output_id, absolute_value):
+        """
+        Set the volume of an output to a 0-100 value.
+
+        Roon endpoints have a few different volume scales - this method scales from 0-100
+        to what the endpoint needs.
+
+        params:
+            output_id: the id of the output
+        """
+        volume_data = self._outputs[output_id].get("volume")
+
+        if volume_data is None:
+            LOGGER.info("This endpoint has fixed volume.")
+            return None
+
+        volume_max = volume_data["max"]
+        volume_min = volume_data["min"]
+        volume_range = volume_max - volume_min
+        volume_percentage_factor = volume_range / 100
+        percentage_volume = volume_min + absolute_value * volume_percentage_factor
+        return self.change_volume_raw(output_id, percentage_volume)
+
+    def change_volume_percent(self, output_id, relative_value):
+        """
+
+        Change the volume of an output by a relative amount.
+
+        Roon endpoints have a few different volume scales - this method scales from 0-100
+        to what the endpoint needs.
+
+        params:
+            output_id: the id of the output
+            relative_value: How much to increase or decrease the volume
+        """
+        volume_data = self._outputs[output_id].get("volume")
+
+        if volume_data is None:
+            LOGGER.info("This endpoint has fixed volume.")
+            return None
+
+        volume_max = volume_data["max"]
+        volume_min = volume_data["min"]
+        volume_range = volume_max - volume_min
+        volume_percentage_factor = volume_range / 100
+
+        volume_percentage_change = int(round(relative_value * volume_percentage_factor))
+        return self.change_volume_raw(output_id, volume_percentage_change, "relative")
+
+    def get_volume_percent(self, output_id):
+        """
+
+        Get the volume of an output.
+
+        Roon endpoints have a few different volumee scales - this method scales from 0-100
+        to what the endpoint needs.
+
+        params:
+            output_id: the id of the output
+            relative_value: How much to increase or decrease the volume
+        """
+
+        volume_data = self._outputs[output_id].get("volume")
+
+        if volume_data is None:
+            LOGGER.info("This endpoint has fixed volume.")
+            return None
+
+        volume_max = volume_data["max"]
+        volume_min = volume_data["min"]
+        volume_range = volume_max - volume_min
+        volume_percentage_factor = volume_range / 100
```

Comment:
This should be `volume_percentage_factor = 100 / volume_range` for getting the value

Reply:
I see that you changed line 310 to a division, so would probably work the way you did it here.

Reply:
Thanks for checking. 

I've added some extra tests - and can confirm this works for my endpoints (which are either (0 to 100) or (-80 to 0)